### PR TITLE
Use body scroll with ha-drawer

### LINKED ISSUE #16031: Lost scrollbar on panel view
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Since beta my front panel view dont show scrollbar to the right. but on the other view that is not panelview it shows. 

### Describe the behavior you expected

image

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

Home Assistant 2023.4.0b5

### What was the last working version of Home Assistant Core?

Home Assistant 2023.3.6

### In which browser are you experiencing the issue with?

Google chrome Version 111.0.5563.147 (Officiell version) (64 bitar)

### Which operating system are you using to run this browser?

Windows 10 Business

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### LINKED ISSUE #16130: 2023.4 Removed Common iOS Gesture Functionality from the iOS App
### The problem

Since the 2023.4 upgrade (and 2023.4.1 and 2023.4.2), the iOS app appears to have lost the ability to tap at the top of the screen to scroll automatically at the top AND it appears it's losing data when the app is the background and the reopened.

Screen Tap - the repo is:
- Open HA App
- Go to a view that is large or long, scroll down to the bottom
- Tap the top of the screen (nothing happens, it previously would scroll all the way to the top)

Reliability - the repro is:
- open HA App
- Navigate away for a bit (maybe 10 min)
- re-open the HA app
- Many of the icons are missing from my main dashboard.  Workaround is to pull down to refresh the page, which restores the icons.  The data appears to be correct, but some of the icons are just missing.

Both of these may be related to the original issue with 2023.4 that removed the left swipe gesture to bring out the menu from the left, which the mobile app update solved almost immediately after 2023.4 was released.

### What version of Home Assistant Core has the issue?

2023.4.2

### What was the last working version of Home Assistant Core?

2023.3.6

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Home Assistant for iOS

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/mobile_app/#apps-that-use-mobile-app

### Diagnostics information

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

Please let me know what specific information I should provide that would be helpful.

### LINKED ISSUE #16154: Add Card button not locked in position moves with dashboard
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When editing dashboard in the HA Android app or mobile browser the add button does not stay locked in the button right corner but instead floats with the dashboard content. 

### Describe the behavior you expected

Add card button should stay locked to the bottom of the screen when editing the dashboard on mobile or android app.

### Steps to reproduce the issue

1. Edit a dashboard that is large enough to scroll on a mobile screen in the Android app or mobile browser.
2.
3.
...

### What version of Home Assistant Core has the issue?

2023.4.3

### What was the last working version of Home Assistant Core?

Unsure

### In which browser are you experiencing the issue with?

Google chrome version 112.0.5615.47

### Which operating system are you using to run this browser?

Android 13

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

Screenshots showing issue

Screenshot_20230413-002207
Screenshot_20230413-002511
Screenshot_20230413-005829

### LINKED ISSUE #16157: Entities card borders are not where they should be
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Since the upgrade to 2023.4.0 (I think), one of my pages is not displaying correctly on any device. I have two entity cards in a horizontal stack. 26 entities in each card, but the border is wrapping the card at the 13th

### Describe the behavior you expected

border to wrap all entities

### Steps to reproduce the issue

please refer to https://community.home-assistant.io/t/entities-card-borders-are-not-where-they-should-be/558913

### What version of Home Assistant Core has the issue?

Home Assistant 2023.4.3

### What was the last working version of Home Assistant Core?

Home Assistant 2023.3.x

### In which browser are you experiencing the issue with?

Firefox, Edge, iOS app, Android App

### Which operating system are you using to run this browser?

Windows, iOS, Android

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Use fixed drawer and fixed top bar to restore body scroll.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #16130, fixes #16154, fixes #16157, fixes #16031
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/hui-root.ts [hunk 1]
```diff
@@ -586,9 +576,12 @@ class HUIRoot extends LitElement {
         constructUrlCurrentPath(removeSearchParam("conversation"))
       );
     }
-    window.addEventListener("scroll", () => {
-      this.toggleAttribute("scrolled", window.scrollY !== 0);
-    });
+    window.addEventListener("scroll", this._handleWindowScroll);
```
Comment (bramkragten) [inline src/panels/lovelace/hui-root.ts:579]: make it a passive listener

**Comment by bramkragten** [comment]
Will fix https://github.com/home-assistant/frontend/issues/16130

**Comment by bramkragten** [comment]
Will fix https://github.com/home-assistant/frontend/issues/16154

**Comment by bramkragten** [comment]
Does it fix https://github.com/home-assistant/frontend/issues/16157 ?

**Comment by piitaya** [comment]
> Does it fix https://github.com/home-assistant/frontend/issues/16157 ?

I have to test that 🙂

**Comment by bramkragten** [comment]
Another one to test: https://github.com/home-assistant/frontend/issues/16031

**Comment by piitaya** [comment]
I can confirm it fixes https://github.com/home-assistant/frontend/issues/16157.
https://github.com/home-assistant/frontend/issues/16031 should be fixed too.

**Review by bramkragten — APPROVED** [review]

**Comment by makhlouta** [comment]
The issue of #16157 is still there after updating to 2023.4.4. But now the border ends at the 20th out of 26 instead of 13th out of 26.

**Comment by BebeMischa** [comment]
Did not happen in 2023.4.5 :-(

**Comment by pisti222** [comment]
> Did not happen in 2023.4.5 :-(

I have same problem, but wersion ugrade does not helped.....

**Comment by piitaya** [comment]
Yeah, this PR is not included in `2023.4.5`. It will be included in the beta next Wednesday (`2023.5.0.b0`).

**Comment by BebeMischa** [comment]
So, for regular users in 2023.5.0, if it passes the beta... pfffff.....

**Pull request merged**